### PR TITLE
Refactor/node list groups

### LINKED ISSUE #2153: Refactor the node-list-groups (also known as filter groups) from the node-list component
When working on the refactoring, please refer to this [miro board](https://miro.com/app/board/uXjVLTeWe88=/) for the discussion from the wider team

### PR DESCRIPTION
**Pull request by Huongg**
## Description
Fixes #2153 

Main changes are as below:
- Separate the `node-list-groups`, `node-list-group`, and `node-list-row-list` from node-list
- Rename them to be "Filters" with the new structure as in the chart below

Screenshot 2024-10-31 at 13 53 22

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by rashidakanchwala — COMMENTED** [review]
Can we have a filters folder, and everything within that filter-group, filter-row, and filter section?

**Comment by Huongg** [comment]
> Can we have a filters folder, and everything within that filter-group, filter-row, and filter section?

hey @rashidakanchwala It is currently in the Filters folder at the moment? Unless you want to move it to somewhere else?
Screenshot 2024-10-31 at 14 12 25

**Comment by rashidakanchwala** [comment]
nice, i got confused coz of the components folder!

File: src/components/filters/components/filters-row/filters-row.js [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+import IndicatorIcon from '../../../icons/indicator';
+import OffIndicatorIcon from '../../../icons/indicator-off';
+import { ToggleControl } from '../../../ui/toggle-control/toggle-control';
+import { RowText } from '../../../ui/row-text/row-text';
 
-import './filter-row.scss';
+import './filters-row.scss';
 
-export const FilterRow = ({
+const FiltersRow = ({
```
Comment (rashidakanchwala) [inline src/components/filters/components/filters-row/filters-row.js:10]: why are we calling it FiltersRow?

**Review by rashidakanchwala — APPROVED** [review]
LGTM. very nice. thanks!

**Review by SajidAlamQB — APPROVED** [review]
Awesome work @Huongg the refactor looks great! 

nit: it might be cleaner to remove the extra `components` folder within filters to keep the directory structure flatter,  wdyt?

**Comment by rashidakanchwala** [comment]
> nit: it might be cleaner to remove the extra components folder within filters to keep the directory structure flatter, wdyt?

I second that !

**Comment by Huongg** [comment]
> Awesome work @Huongg the refactor looks great!
> 
> nit: it might be cleaner to remove the extra `components` folder within filters to keep the directory structure flatter, wdyt?

I don't mind I can remove the components folder 👍

**Comment by jitu5** [comment]
@Huongg 
[nit] Two minor observation

1. For tag,  group toggle icon is always visible compared to demo site 
2. Selected blue color is little different compared to demo site  
Screenshot 2024-11-01 at 3 59 40 p m

Demo site
Screenshot 2024-11-01 at 4 00 38 p m

**Review by jitu5 — APPROVED** [review]
This is really clean, thanks @Huongg.

**Comment by Huongg** [comment]
> @Huongg [nit] Two minor observations.
> 
> 1. For tag,  group toggle icon is always visible compared to demo site
> 2. Selected blue color is little different compared to demo site
> 
 hey @jitu5 what you see in the demo site is a bug, the box should always visible, so I fixed it in this PR now

**Pull request merged**